### PR TITLE
[Merged by Bors] - chore: use `FunLike` for `OrderHom`

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #5820

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
@jjaassoonn I'm not sure that silencing linter is the best way to go. We have lots of `rfl` simp lemmas (a.k.a. `dsimp` lemmas) and Lean matches LHS based on a discrimination tree, so it may fail to apply this lemma after "simplifying" LHS.

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5820
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Topology/Order/Hom/Esakia.lean [hunk 1]
```diff
@@ -122,6 +122,8 @@ instance : CoeFun (PseudoEpimorphism α β) fun _ => α → β :=
   FunLike.hasCoeToFun
 
 @[simp]
+theorem toOrderHom_eq_coe (f : PseudoEpimorphism α β) : ⇑f.toOrderHom = f := rfl
+
 theorem toFun_eq_coe {f : PseudoEpimorphism α β} : f.toFun = (f : α → β) := rfl
```
Comment (eric-wieser) [inline Mathlib/Topology/Order/Hom/Esakia.lean:127]:
```suggestion
@[simp]
theorem toFun_eq_coe {f : PseudoEpimorphism α β} : f.toFun = (f : α → β) := rfl
```

File: Mathlib/Order/OmegaCompletePartialOrder.lean [hunk 2]
```diff
@@ -109,10 +110,13 @@ variable (g : β →o γ)
 instance : LE (Chain α) where le x y := ∀ i, ∃ j, x i ≤ y j
 
 /-- `map` function for `Chain` -/
-@[simps! (config := { fullyApplied := false })]
+-- Porting note: `simps` doesn't work with type synonyms
+-- @[simps! (config := { fullyApplied := false })]
```
[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/Order/OmegaCompletePartialOrder.lean:114]: Before, coercion to function used `toFun`. For some reason, `simps`/`simp` can work with it but not with `FunLike.coe`.

File: Mathlib/Topology/Order/Hom/Esakia.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Topology/Order/Hom/Esakia.lean:127]: No, the previous lemma can prove it.

**Comment by ocfnash** [comment]
bors merge

**Comment by ocfnash** [comment]
bors d+

**Comment by urkud** [comment]
It was a glitch with cache, trying again
bors merge

**Pull request closed without merging**